src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.ParseResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame
+{
+    internal partial struct StackFrameParser
+    {
+        private struct ParseResult<T>
+        {
+            public readonly bool Success;
+            public readonly T? Value;
+
+            public static readonly ParseResult<T> Abort = new(false, default);
+            public static readonly ParseResult<T> Empty = new(true, default);
+
+            public ParseResult(T? value)
+                : this(true, value)
+            { }
+
+            private ParseResult(bool success, T? value)
+            {
+                Success = success;
+                Value = value;
+            }
+
+            public void Deconstruct(out bool success, out T? value)
```

Comment:
consider making this a a record struct.  then you get this for free.

Reply:
I didn't make this a record just because it makes controlling the constructors a bit cleaner. Now the only way to construct is either pass a specific `Value` or use `Abort` or `Empty`

Reply:
ah, you're right.  good point.